frontend/src/app/components/checkin/check-in-form/check-in-form.component.ts

```diff
@@ -56,15 +59,22 @@ export class CheckInFormComponent implements OnInit {
     this.dialogForm.controls.actionList.setValue(this.keyResult.actionList);
     if (this.data.checkIn != null) {
       this.checkIn = this.data.checkIn;
-      this.dialogForm.controls.value.setValue(this.checkIn.value!.toString());
+      if ((this.checkIn as CheckInMetricMin).value != null) {
+        this.dialogForm.controls.value.setValue(this.getCheckInMetric().value!.toString());
+      } else {
+        this.dialogForm.controls.value.setValue(this.getCheckInOrdinal().zone!);
+      }
```

Comment:
The `getCheckInMetric` and `getCheckInOrdinal` methods are only used here, so this logic can be moved to a separate method alltogether, i.e. `getCheckInValue`.

This would then simplify the code here back to something like this:
```
this.dialogForm.controls.value.setValue(this.getCheckInValue());
```